src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -224,9 +288,20 @@ protected internal override async Task<HttpResponseMessage> SendAsync(HttpReques
                 JSObject t = (JSObject)await response.ConfigureAwait(continueOnCapturedContext: true);
 
                 var status = new WasmFetchResponse(t, abortController, abortCts, abortRegistration);
-
                 HttpResponseMessage httpResponse = new HttpResponseMessage((HttpStatusCode)status.Status);
 
+                // Here we will set the ReasonPhrase so that it can be evaluated later.
+                // We do not have a status code but this will signal some type of what happened
+                // after interrogating the status code for success or not i.e. IsSuccessStatusCode
+                //
+                // https://developer.mozilla.org/en-US/docs/Web/API/Response/type
+                // opaqueredirect: The fetch request was made with redirect: "manual".
+                // The Response's status is 0, headers are empty, body is null and trailer is empty.
+                if (!string.IsNullOrEmpty(status.ResponseType) && status.ResponseType == "opaqueredirect")
```

Comment:
Why check for `!string.IsNullOrEmpty(status.ResponseType)` ? Shouldn't this be enough?

```suggestion
                if (status.ResponseType == "opaqueredirect")
```